app/models/concerns/course/levels_concern.rb

```diff
@@ -41,4 +41,35 @@ def compute_level_number(experience_points)
     level = compute_level(experience_points)
     level ? level.level_number : 0
   end
+
+  # Returns the current level's experience point (exp) threshold for
+  # for a given number of experience_points.
```

Comment:
Chatted with @kxmbrian and read through what he considered when designing EXP and Levels [here](https://github.com/Coursemology/coursemology2/wiki/EXP-Points-and-Levels), and realised that it was designed to have a _Level 0_. Current this is handled by the model and level 0 returns a `nil` object.

There are 3 options moving forward:
 1. Each `course` to store a default `level 0` - will need some work to maintain this. 
 2. `Levels` model to return a `Levels` object if `CourseUser` is level 0.
 3. Implement `CourseUser.current_level` and `CourseUser.next_level` to handle the edge cases. 

Option 1 seem cleanest to me, will prefer to go with that that. 

Another issue: Currently, `Levels` thresholds are set to be strictly positive, while EXP can be negative. Ideally the Levels threshold should match to the EXP space (so if EXP can be negative, levels can be negative also). If we go with Option 1, Level 0's threshold can be negative infinity (or some max number we define). 

Does that sound okay?